cukinia

```diff
@@ -680,6 +681,73 @@ _cukinia_systemd_failed()
 	fi
 }
 
+# _cukinia_gpio_libgpiod: Check gpio configuration via libgpiod
+# arg1: (optional) the pins configured in input
+# arg2: (optional) the pins configured in output high
+# arg3: (optional) the pins configured in output low
+# arg4: (optional) the gpiochip used (default:gpiochip0)
+_cukinia_gpio_libgpiod()
+{
+	local input_pins
+	local output_low_pins
+	local output_high_pins
+	local gpiochip="gpiochip0"
+	local direction
+	local value
+
+	while [ $# -gt 0 ]; do
+		case "$1" in
+		-i)
+			input_pins="$2"
+			shift
+			;;
+		-l)
+			output_low_pins="$2"
+			shift
+			;;
+		-h)
+			output_high_pins="$2"
+			shift
+			;;
+		-g)
+			gpiochip="$2"
+			shift
+			break
+			;;
+		esac
+		shift
+	done
+
+	_cukinia_prepare "Checking if gpio pins are well configured via libgpiod"
```

Comment:
does that work with dts-named gpios ? for example with something like

```
	gpio-line-names = "FRAME_ON_N",
			  "NC",
			  "NC",
```

Reply:
yes it will work because  the validation is based on line number. In your example gpioinfo will display for example:
`line   0:      FRAME_ON_N     unused   input  active-high 
line   1:      unnamed       unused   input  active-high 
line   2:      unnamed       unused   input  active-high`
So naming a gpio will not break the configuration test.